labs/intro-to-ada/chapters/solutions/standard_library.rst

```diff
@@ -0,0 +1,1031 @@
+       is
+          ID : Item_ID;
+       begin
+          for C in Inv.List_Item.Iterate loop
+             if To_String (Inv.List_Item (C).Name) = Item_Name then
+                ID := Item_ID (C);
+                exit;
+             end if;
+          end loop;
+
+          return ID;
+       end Get;
+
+       procedure Set (Inv      : in out Inventory;
+                      Trans    :        Transaction_Type;
+                      ID       :        Item_ID;
+                      Quantity :        Positive;
+                      Success  :    out Boolean)
+       is
+          Q : Integer;
+          C : constant Item_C := Item_C (ID);
+       begin
+          case Trans is
+             when Bought =>
+                Q := Inv.List_Item (C).Stock_Quantity + Quantity;
+             when Sold =>
+                Q := Inv.List_Item (C).Stock_Quantity - Quantity;
+          end case;
+
+          if Q >= 0 then
+             Success := True;
+
+             Inv.List_Item (C).Stock_Quantity := Q;
+
+             Inv.List_Item (C).Stock_Amount :=
+               Amount (Q) * Inv.List_Item (C).Price;
+
+             Inv.List_Item (C).Trans_Quantity (Trans) :=
+               Inv.List_Item (C).Trans_Quantity (Trans) + Quantity;
+
+             Inv.List_Item (C).Trans_Amount (Trans) :=
+               Inv.List_Item (C).Trans_Amount (Trans) +
+               Amount (Quantity) * Inv.List_Item (C).Price;
+          else
+             Success := False;
+          end if;
+
+       end Set;
+
+       function Get (Inv   : Inventory;
+                     ID    : Item_ID) return String is
+         (To_String (Inv.List_Item (Item_C (ID)).Name));
+
+       function Get (Inv   : Inventory;
+                     ID    : Item_ID) return Item_Quantity is
+         (Inv.List_Item (Item_C (ID)).Stock_Quantity);
+
+       function Get (Inv   : Inventory;
+                     ID    : Item_ID) return Amount is
+         (Inv.List_Item (Item_C (ID)).Stock_Amount);
+
+       function Get (Inv   : Inventory;
+                     Trans : Transaction_Type;
+                     ID    : Item_ID) return Item_Quantity is
+         (Inv.List_Item (Item_C (ID)).Trans_Quantity (Trans));
+
+       function Get (Inv   : Inventory;
+                     Trans : Transaction_Type;
+                     ID    : Item_ID) return Amount is
+         (Inv.List_Item (Item_C (ID)).Trans_Amount (Trans));
+
+       function Get (Inv   : Inventory;
+                     Trans : Transaction_Type) return Amount
+       is
+          Total : Amount := 0.0;
+       begin
+          for C in Inv.List_Item.Iterate loop
+             Total := Total + Get (Inv, Trans, Item_ID (C));
+          end loop;
+
+          return Total;
+       end Get;
+
+       function Get (Inv   : Inventory) return Amount
+       is
+          Total : Amount := 0.0;
+       begin
+          for C in Inv.List_Item.Iterate loop
+             Total :=  + Get (Inv, Item_ID (C));
```

Comment:
If someone looks at the solutions for help, do we want to add comments when things are confusing? I thought line 349 was an error, but it's just using OOP notation. But we don't do that on line 338. So maybe a comment indicating that?

Reply:
Thanks for noticing this: it was a bug. I've just fixed it.